trieste/acquisition/sampler.py

```diff
@@ -269,6 +269,10 @@ def __init__(self, sample_size: int, model: ProbabilisticModel):
             tf.ones([0, 0, sample_size], dtype=tf.float64), shape=[None, None, sample_size]
         )  # [0, 0, S]
 
+        # for some reason graph compilation is resulting in self._eps reporting the wrong shape
+        # we therefore use an extra boolean variable to keep track of whether it's initialised
+        self._initialized = tf.Variable(False)
```

Comment:
Why does this have to be a tf Variable?

Reply:
For the same reason self._eps needs to be: so it can be updated in the compiled graph.